modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/FilterToSqlHelper.java

```diff
@@ -263,8 +264,13 @@ void visitComparisonSpatialOperator(BinarySpatialOperator filter,
         
         // add && filter if possible
         if(!(filter instanceof Disjoint)) {
-            
+            if(encodeBBOXFilterAsEnvelope && !isCurrentGeography()) {
+                out.write("ST_envelope(");
```

Comment:
This may be an internal detail of postgis? Perhaps I have my naming convention wrong ...